watertap/unit_models/clarifier.py

```diff
@@ -0,0 +1,148 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Clarifier unit model for BSM2 and plant-wide wastewater treatment modeling.
+This unit inherits from the IDAES separator unit.
+"""
+from enum import Enum, auto
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+)
+from idaes.models.unit_models.separator import SeparatorData, SplittingType
+
+from idaes.core.util.tables import create_stream_table_dataframe
+import idaes.logger as idaeslog
+
+from pyomo.environ import (
+    Var,
+    Param,
+    units as pyunits,
+)
+
+from idaes.core.util.exceptions import (
+    ConfigurationError,
+)
+from watertap.costing.unit_models.clarifier import cost_clarifier
+
+__author__ = "Chenyu Wang"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+class ActivatedSludgeModelType(Enum):
+    """
+    ASM1: ASM1 model
+    ASM2D: ASM2D model
+    modified_ASM2D: modified ASM2D model for ADM1 compatibility
+    """
+
+    ASM1 = auto()
+    ASM2D = auto()
+    modified_ASM2D = auto()
+
+
+@declare_process_block_class("Clarifier")
+class ClarifierData(SeparatorData):
+    """
+    Thickener unit model for BSM2
+    """
+
+    CONFIG = SeparatorData.CONFIG()
+    CONFIG.outlet_list = ["underflow", "overflow"]
+    CONFIG.split_basis = SplittingType.componentFlow
+
+    def build(self):
+        """
+        Begin building model.
+        Args:
+            None
+        Returns:
+            None
+        """
+
+        # Call UnitModel.build to set up dynamics
+        super(ClarifierData, self).build()
+
+        if "underflow" and "effluent" not in self.config.outlet_list:
+            raise ConfigurationError(
+                "{} encountered unrecognised "
+                "outlet_list. This should not "
+                "occur - please use underflow "
+                "and effluent as outlets.".format(self.name)
+            )
+
+        self.surface_area = Var(
+            initialize=1500,
+            doc="Cross section surface area of the clarifier",
+            units=pyunits.m**2,
+            bounds=(0, 3000),
```

Comment:
Where does this upper bound come from? For example, the cost curve that I found for gravity sludge thickeners provides cost related to the diameter of the thickener, and the stated valid range for diameter is 20 - 50 feet. Did your upper bound originate from a similar valid range stated in the literature or did you just assume this?

Reply:
This upper bound is also from the same source literature

Reply:
Shouldn't this be 200 sq ft, based on what I see in the costing file?

Reply:
the 200 sq ft is only for O&M cost, the construction cost relationship is valid up to 32000 sq ft